apps/server/src/shared/infra/mail/mail.service.ts

```diff
@@ -16,6 +17,39 @@ export class MailService {
 	) {}
 
 	public async send(data: Mail): Promise<void> {
+		if (Configuration.has('ADDITIONAL_BLACKLISTED_EMAIL_DOMAINS')) {
+			const domainBlockList = Configuration.get('ADDITIONAL_BLACKLISTED_EMAIL_DOMAINS') as string;
+			data.recipients = this.removeEmailAddressesThatHaveBlockedDomain(data.recipients, domainBlockList) as string[];
+			data.cc = this.removeEmailAddressesThatHaveBlockedDomain(data.cc, domainBlockList);
+			data.bcc = this.removeEmailAddressesThatHaveBlockedDomain(data.bcc, domainBlockList);
+			data.replyTo = this.removeEmailAddressesThatHaveBlockedDomain(data.replyTo, domainBlockList);
+		}
+
+		if (data.recipients.length === 0) {
+			return;
+		}
+
 		await this.amqpConnection.publish(this.options.exchange, this.options.routingKey, data, { persistent: true });
 	}
+
+	private removeEmailAddressesThatHaveBlockedDomain(
+		mails: string[] | undefined,
+		domainBlockList: string
+	): string[] | undefined {
+		if (mails === undefined || mails === null) {
+			return mails;
+		}
+		const domainBlockListArray = domainBlockList.split(',').map((domain) => domain.trim());
+
+		for (const blockedDomain of domainBlockListArray) {
```

Comment:
Please revert the order - don't go through every blacklisted domain first, but through every email address and then check if its domain is included in the blacklisted domain list - it's easier to read and better performance-wise - there can be no need to go through every blacklisted domain.

Reply:
done